### PR TITLE
modified traincontroller implementation

### PR DESCRIPTION
**Pull request by molnartimi**

### CODE REVIEW

**Review by zongolista — CHANGES_REQUESTED** [review]
timer should wait 1000 millisec

**Review by zongolista — APPROVED** [review]

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java [hunk 1]
```diff
@@ -22,6 +25,20 @@ public void followSpeed() {
...
 		enforceSpeedLimit();
 	}
+	
+	public void changeReferenceSpeed() {
+		if (timer == null) {
+			timer = new Timer();
+		}
+		while (true) {
+			try {
+				timer.wait(500);
```
Comment (molnartimi) [inline train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java]: It should be 1000.

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java [hunk 2]
```diff
@@ -22,6 +25,20 @@ public void followSpeed() {
...
+		if (timer == null) {
+			timer = new Timer();
+		}
+		while (true) {
+			try {
+				timer.wait(1000);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			followSpeed();
```
Comment (zongolista) [inline train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java]: followSpeed() should be called before try{}

**Review by zongolista — APPROVED** [review]
okay :(

**Pull request merged**